### DISCUSSION TITLE
3rd party libraries - updates and requests

### DISCUSSION THREAD

**Discussion by cortinico** [reactwg/react-native-new-architecture#6] [Libraries]
### Please refer to the new discussion: https://github.com/reactwg/react-native-new-architecture/discussions/167

---

Hi all,
It's crucial that **3rd party libraries** and tools for React Native are supporting the New React Native Architecture, both Fabric and TurboModules.

## Known Libraries Status

| library                        | status                  | relevant links                                                                                                                    |
| ------------------------------ | ----------------------- | --------------------------------------------------------------------------------------------------------------------------------- |
| react-native-keys        | completed             | https://github.com/numandev1/react-native-keys/releases/tag/v0.6.0-alpha.0
| react-native-compressor        | completed             | https://github.com/numandev1/react-native-compressor/releases/tag/v1.7.0
| react-native-bottom-sheet      | not started             | https://github.com/gorhom/react-native-bottom-sheet/issues/1010 / https://github.com/gorhom/react-native-bottom-sheet/issues/1030 |
| react-native-fast-image        | not started             | https://github.com/DylanVann/react-native-fast-image/issues/919                                                                   |
| react-native-device-info        | not started             | https://github.com/react-native-device-info/react-native-device-info/issues/1470                                                                   |
| react-native-navigation        | not started             | https://github.com/wix/react-native-navigation/issues/7466                                                                        |
| react-native-permissions        | not started             | https://github.com/zoontek/react-native-permissions/issues/684                                                                        |
| react-native-haptic-feedback        | not started             | https://github.com/junina-de/react-native-haptic-feedback/issues/87                                                                        |
| react-native-pure-jwt        | not started             | https://github.com/zaguiini/react-native-pure-jwt/issues/41                                                                        |
| react-native-pdf               | not started             | https://github.com/wonday/react-native-pdf/issues/663                                                                             |
| react-native-PDFView           | not started             | https://github.com/rumax/react-native-PDFView/issues/229                                                                          |
| react-native-shimmer           | not started             | https://github.com/oblador/react-native-shimmer/issues/35                                                                         |
| react-native-firebase           | not started             | https://github.com/invertase/react-native-firebase/discussions/6200                                                                         |
| react-native-skia              | not started             | https://github.com/Shopify/react-native-skia/issues/387                                                                           |
| react-native-video             | not started             | https://github.com/react-native-video/react-native-video/issues/2650                                                              |
| react-native-maps              | investigation           | https://github.com/react-native-maps/react-native-maps/issues/4383                                                                |
| sentry-react-native            | investigation           | https://github.com/getsentry/sentry-react-native/issues/2226                                                                      |
| react-native-picker    | in progress             | https://github.com/react-native-picker/picker/pull/456                                                          |
| react-native-datetimepicker    | in progress             | https://github.com/react-native-datetimepicker/datetimepicker/issues/622                                                          |
| lottie-react-native            | complete             | https://github.com/lottie-react-native/lottie-react-native/releases/tag/v6.0.0                                                               |
| react-native-blurhash          | in progress             | https://github.com/mrousavy/react-native-blurhash/pull/148                                                                        |
| react-native-masked-view          | in progress             | https://github.com/react-native-masked-view/masked-view/issues/162                                                                        |
| react-native-vision-camera     | in progress             | https://github.com/mrousavy/react-native-vision-camera/pull/650                                                                   |
| react-native-webview           | RC ongoing             | https://github.com/react-native-webview/react-native-webview/pull/2686                                                            |
| react-native-netinfo           | in progress             | https://github.com/react-native-netinfo/react-native-netinfo/pull/622                                                            |
| pendo-sdk           | in progress             | https://github.com/pendo-io/pendo-mobile-ios                                                                                  |
| react-native-gesture-handler   | complete                | https://github.com/software-mansion/react-native-gesture-handler/releases/tag/2.3.0                                               |
| react-native-linear-gradient   | (_forked but_) complete | https://github.com/FyndX/react-native-gradient                                                                                    |
| react-native-pager-view        | complete             | https://github.com/callstack/react-native-pager-view/releases/tag/v6.1.0-rc.1                                                                     |
| react-native-reanimated        | complete                | https://github.com/software-mansion/react-native-reanimated/releases/tag/3.0.0-rc.0                                               |
| react-native-safe-area-context | complete                | https://github.com/th3rdwave/react-native-safe-area-context/releases/tag/v4.1.2                                                   |
| react-native-screens           | complete                | https://github.com/software-mansion/react-native-screens/releases/tag/3.12.0                                                      |
| react-native-slider            | complete                | https://github.com/callstack/react-native-slider/releases/tag/v4.3.0                                                              |
| react-native-svg               | complete                | https://github.com/react-native-svg/react-native-svg/releases/tag/v13.0.0                                                         |
| react-native-avoid-softinput              | complete                | https://github.com/mateusz1913/react-native-avoid-softinput/releases/tag/v3.0.0                                                         |
| react-native-blur            | complete                | https://github.com/Kureev/react-native-blur/pull/493                                                         |
| react-native-pushy        | complete                | https://github.com/reactnativecn/react-native-pushy/pull/402                                           |

## For Library Authors/Maintainers

If you're a library author/maintainer, feel free **to open a discussion in this section** to provide updates on the status of your library. 

If you happen to have a Alpha/Beta/RC ready to share, feel free to do so. You can use this working group for getting visibility and early feedback.

Please open only **one thread per library**.

## For Library User

If you're a library user, **feel free to comment here** if a library you're depending on it's not migrated yet. 

This will help the library author/maintainer collect use cases and will help them prioritize the migration work.

**Comment by tido64** [comment]
@cortinico: Would this be the place to discuss adding support for autolinking TurboModules in [`@react-native-community/cli`](https://github.com/react-native-community/cli)?

**Reply by thymikee** [reply]
I'd be interested to know what we can do from the RN CLI side

**Reply by cortinico** [reply]
I'd say let's either move to https://github.com/reactwg/react-native-new-architecture/discussions/2 as it's a technical implementation detail to TM. If you feel this could get bigger, feel free to kickoff a discussion inside Libraries 👍 

I also have some relevant context to share on TM Autolinking so looking forward to this :)

**Reply by tido64** [reply]
Moved here: https://github.com/reactwg/react-native-new-architecture/discussions/2#discussioncomment-2388444

**Comment by mk0116** [comment]
https://github.com/react-native-video/react-native-video
react-native-video needs to be migrated.
If help is needed, I can do some work for it.

**Reply by cortinico** [reply]
I've opened this https://github.com/react-native-video/react-native-video/issues/2650 to understand what's the status. If you're willing to contribute, please follow up over there so the maintainers are aware of it.

**Reply by mk0116** [reply]
Thanks!

**Comment by zolbooo** [comment]
It would be nice to have some kind of SVG support, for example: `react-native-svg` (there is a draft PR in progress).
As alternative, we use `react-native-vector-image` with fork of [`@klarna/react-native-vector-drawable`](https://github.com/zolbooo/react-native-vector-drawable) so far.

**Comment by kelset** [comment]
quick suggestion - maybe could be worth to have a kind of table in the post with a at-a-glance overview of the state based on what folks are reporting?
I've jotted down the state for now and thi seems to be the state:
- react-native-video = not started ([https://github.com/react-native-video/react-native-video/issues/2650](https://github.com/react-native-video/react-native-video/issues/2650))
- react-native-navigation = not started ([https://github.com/wix/react-native-navigation/issues/7466](https://github.com/wix/react-native-navigation/issues/7466))
- react-native-slider = in progress ([https://github.com/callstack/react-native-slider/issues/380](https://github.com/callstack/react-native-slider/issues/380), https://github.com/callstack/react-native-slider/pull/400)
- react-native-webview = in progress ()
- react-native-pager-view = in progress ([https://github.com/callstack/react-native-pager-view/pull/534](https://github.com/callstack/react-native-pager-view/pull/534))
- react-native-safe-area-context = complete ([https://github.com/th3rdwave/react-native-safe-area-context/releases/tag/v4.1.2](https://github.com/th3rdwave/react-native-safe-area-context/releases/tag/v4.1.2))
- react-native-screens = complete ([https://github.com/software-mansion/react-native-screens/releases/tag/3.12.0](https://github.com/software-mansion/react-native-screens/releases/tag/3.12.0))
- react-native-gesture-handler = complete ([https://github.com/software-mansion/react-native-gesture-handler/releases/tag/2.3.0](https://github.com/software-mansion/react-native-gesture-handler/releases/tag/2.3.0))
- react-native-reanimated = complete ([https://github.com/software-mansion/react-native-reanimated/releases/tag/3.0.0-rc.0](https://github.com/software-mansion/react-native-reanimated/releases/tag/3.0.0-rc.0))
- react-native-linear-gradient = forked but complete ([https://github.com/FyndX/react-native-gradient](https://github.com/FyndX/react-native-gradient))
- react-native-skia = not started (https://github.com/Shopify/react-native-skia/issues/387)
- react-native-svg = ongoing (https://github.com/react-native-svg/react-native-svg/pull/1754 / https://github.com/react-native-svg/react-native-svg/pull/1804)

**Reply by leotm** [reply]
all-in-one place looks good ^ kinda like we did in the [blog](https://reactnative.dev/blog/2022/06/16/resources-migrating-your-react-native-library-to-the-new-architecture#migration-status-of-popular-libraries)
and one more maybe to add
- react-native-skia = not started (https://github.com/Shopify/react-native-skia/issues/387)

**Reply by kelset** [reply]
oh THAT'S WHY! I was so sure that a list like this existed somewhere but couldn't find it anymore 🤦‍♂️ it was in the blogpost 🤦‍♂️ thanks @leotm 

if @cortinico gives me the 👍 I'll just edit the top comment and make it into a nice table :3

**Reply by kelset** [reply]
(added to the top post, added skia and svg)

**Comment by capezzbr** [comment]
One more library to add: https://github.com/lottie-react-native/lottie-react-native. Migration has started for Android: https://github.com/lottie-react-native/lottie-react-native/pull/910

**Reply by kelset** [reply]
thanks!

**Comment by capezzbr** [comment]
Any PDF library: https://github.com/rumax/react-native-PDFView or https://github.com/wonday/react-native-pdf

**Reply by kelset** [reply]
sorry I'm not sure I follow - are these libs undergoing migration to the new arch or already support the new arch?

**Reply by capezzbr** [reply]
hey, unfortunately from what I can tell they haven't started any migration yet. With the above comment I mean that I'd be useful to track the progress of migration of at least one library that allows us to render a PDF 👍

**Comment by cipolleschi** [comment]
[react-native-PDFView](https://github.com/rumax/react-native-PDFView) does not support it.

[react-native-pdf](https://github.com/wonday/react-native-pdf) has [an issue](https://github.com/wonday/react-native-pdf/issues/663) to support it, but no PR and no work so far.

I think this is a request to migrate one of these two.

**Comment by capezzbr** [comment]
I have two more libraries that I'd like to track:
- `@gorhom/bottom-sheet` -- https://github.com/gorhom/react-native-bottom-sheet
- `@react-native-community/datetimepicker` -- https://github.com/react-native-datetimepicker/datetimepicker

**Reply by kelset** [reply]
@capezzbr for the future: can you check if it already exist or add an issue on those repos about new arch support and link it here along with the track requests?

**Reply by capezzbr** [reply]
Yes! Let me update it right now:
- For `@gorhom/bottom-sheet` I did create https://github.com/gorhom/react-native-bottom-sheet/issues/1010, but the issue was closed due to (30 days) inactivity
- For `@react-native-community/datetimepicker` I did create https://github.com/react-native-datetimepicker/datetimepicker/issues/622

**Comment by guptaprnv** [comment]
pls include these library also
1 react-native-interactable 
2 for react-native-shimmer i have created ticket (https://github.com/oblador/react-native-shimmer/issues/35)

**Reply by numandev1** [reply]
@guptaprnv done

**Comment by numandev1** [comment]
I think we should add this discussion URL into react native docs so everyone may check new arch-supported libraries status before their project upgradation

**Comment by intergalacticspacehighway** [comment]
I've started working on turbomodule migration of `@react-native-community/netinfo` [here](https://github.com/intergalacticspacehighway/react-native-netinfo/tree/turbomodule-migration). It's working on android and iOS, also backward compatible 🥳. I'll make a PR soon after some more testing. Some issues I faced are reported [here](https://github.com/reactwg/react-native-new-architecture/discussions/68#discussioncomment-3541788) and [here](https://github.com/reactwg/react-native-new-architecture/discussions/68#discussioncomment-3561006)

**Reply by intergalacticspacehighway** [reply]
PR - https://github.com/react-native-netinfo/react-native-netinfo/pull/622

**Comment by dmahajan980** [comment]
Hello, it would be nice to track `react-native-firebase`: https://github.com/invertase/react-native-firebase
A discussion was opened in the repo regarding Fabric / TurboModules support: https://github.com/invertase/react-native-firebase/discussions/6200

**Reply by kelset** [reply]
added, thanks

**Comment by hansdesmedt** [comment]
react-native-device-info (not started)
https://github.com/react-native-device-info/react-native-device-info/issues/1470
Great lib we use a lot. 😄

react-native-permissions (not started)
zoontek is asking for help to update his libraries
https://github.com/sponsors/zoontek?frequency=one-time
we use [react-native-permissions](https://github.com/zoontek/react-native-permissions) a lot.

react-native-haptic-feedback (not started)
classic
https://github.com/junina-de/react-native-haptic-feedback/issues/87

react-native-pure-jwt (not started)
one of the rare old ones
https://github.com/zaguiini/react-native-pure-jwt/issues/41

react-native-masked-view (in progress)
was looking into this on myself
https://github.com/react-native-masked-view/masked-view/issues/162
https://github.com/reactwg/react-native-new-architecture/discussions/80

**Reply by kelset** [reply]
is there an issue/pr tracking work on the new arch? I couldn't see anything

**Reply by hansdesmedt** [reply]
I can make one for all of them if needed

**Reply by hansdesmedt** [reply]
I added the issues 😄

**Reply by kelset** [reply]
awesome, thanks Hans! I'll update the top table right now

**Comment by mateusz1913** [comment]
👋  I recently added a Fabric/TurboModules support in react-native-avoid-softinput from [v3.0.0](https://github.com/mateusz1913/react-native-avoid-softinput/releases/tag/v3.0.0) onwards

**Comment by gispada** [comment]
Support for Fabric has also been added to [react-native-blur](https://github.com/Kureev/react-native-blur). See [this PR](https://github.com/Kureev/react-native-blur/pull/493).

**Reply by kelset** [reply]
great stuff, thanks for letting us know 🙏

**Comment by TheTushar2696** [comment]
Hi Everyone,
What if a 3rd party library is not supporting new architecture, whereas my project does, will that library won't work, or its just that we won't be able to leverage the features of the new arch?

**Reply by kelset** [reply]
good question; I've investigated this topic a bit and this is how the situation looks like, currently: https://github.com/kelset/react-native-new-arch-limits

**Reply by TheTushar2696** [reply]
Hi @kelset ,

Thanks for the insights, I also tried some libraries, not visual that were not supporting new architecture, but It was working in the new architecture enabled in my project, We are planning to migrate our legacy code to the new architecture but still not able to figure out what will work and what will not because of the above mentioned question.

**Reply by TheTushar2696** [reply]
Hi @kelset ,

Thanks for the insights, I also tried some libraries, not visual that were not supporting new architecture, but It was working in the new architecture enabled in my project, We are planning to migrate our legacy code to the new architecture but still not able to figure out what will work and what will not because of the above mentioned question.
@cortinico  Any inputs from your end on this?

**Reply by kelset** [reply]
> I also tried some libraries, not visual that were not supporting new architecture, but It was working in the new architecture enabled in my project

if you look at the table in my repo, you'll see that what you are finding is consistent with my findings. Those will work.

**Reply by TheTushar2696** [reply]
> > I also tried some libraries, not visual that were not supporting new architecture, but It was working in the new architecture enabled in my project
> 
> if you look at the table in my repo, you'll see that what you are finding is consistent with my findings. Those will work.

@kelset  And what about NativeModules those are coming as null, will we have to move all of those to turbo modules? Do you have any say on that?

**Comment by mrousavy** [comment]
Hey @cortinico, thanks for the awesome sum up.

JFYI: RN Skia is in progress: https://github.com/Shopify/react-native-skia/pull/937

**Comment by mrousavy** [comment]
react-native-vision-camera and react-native-blurhash are written in Swift, and since Fabric Views use C++ code (.mm) it's probably gonna be a bit tricky to add Fabric support to my libs as of right now since Swift cannot interop with C++.

To unblock me (and other Swift libs), I think it'd make sense to either

1. Add a Swift/extern C wrapper to expose that to Swift, for example for the
    ```mm
    - updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps;
    ```
    function, where I get the props in a `NSDictionary` (ObjC bridged type), or a C-style array instead - or

2. Perform a lookup at runtime.
    Instead of hardcoding it like this:

    ```cpp
    - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
     {
       const auto &oldViewProps = *std::static_pointer_cast<const BlurViewProps>(_props);
       const auto &newViewProps = *std::static_pointer_cast<const BlurViewProps>(props);
    
       if (oldViewProps.blurAmount != newViewProps.blurAmount) {
         NSNumber *blurAmount = [NSNumber numberWithInt:newViewProps.blurAmount];
         [self setBlurAmount:blurAmount];
       }

       if (oldViewProps.blurType != newViewProps.blurType) {
         NSString *blurType = [NSString stringWithUTF8String:toString(newViewProps.blurType).c_str()];
         [self setBlurType:blurType];
       }

       if (oldViewProps.reducedTransparencyFallbackColor != newViewProps.reducedTransparencyFallbackColor) {
         UIColor *color = RCTUIColorFromSharedColor(newViewProps.reducedTransparencyFallbackColor);
         [self setReducedTransparencyFallbackColor:color];
       }

       [super updateProps:props oldProps:oldProps];
     }
    ```

    We could have the default implementation be 
    ```cpp
    - (void)updateProps:(Props::Shared const &)props oldProps:(Props::Shared const &)oldProps
     {
       const auto &oldViewProps = *std::static_pointer_cast<const BlurViewProps>(_props);
       const auto &newViewProps = *std::static_pointer_cast<const BlurViewProps>(props);
    
       // 1. loop over each prop
       // 2. compare if it changed, and if yes find a selector that updates this prop (by converting from C++ to ObjC types)

       [super updateProps:props oldProps:oldProps];
     }
    ```


wdyt?

**Reply by cipolleschi** [reply]
Hi @mrousavy, thank you for the suggestions.

I think that we can try something similar to a certain extent. However, I'm a bit worried concerned because C++ has various way to define its properties: you can have const and pointers, for example, and different kind of pointers.

I can't be tricky to automatically generate a functioning code for all these combinations. Plus, as you can see already with your example, we should have to hardcode in the Codegen the various conversions. For example:
```objc
[NSNumber numberWithInt:newViewProps.blurAmount];
[NSString stringWithUTF8String:toString(newViewProps.blurType).c_str()];
RCTUIColorFromSharedColor(newViewProps.reducedTransparencyFallbackColor);
```
This adds complexity and I think that these two are (at least) part of the reasons why we haven't codegen'd an automatic prop update.

PErhaps @sammy-SC has more insights on the problem.

**Reply by sammy-SC** [reply]
@cipolleschi that's a good summary. I think you mentioned all the important points. 

@mrousavy 
Question about the 1st approach, how would we get `NSDictionary` from old and new props struct? What about types which are stored in struct signature? 
For the 2nd approach, how would we build the looping mechanism? As far as I know, C++ standard does not provide a way to loop over data members. How would we know the type of data member inside the loop?


it has been a while since I looked into [C++ <-> Swift interoperability](https://github.com/apple/swift/blob/main/docs/CppInteroperability/CppInteroperabilityStatus.md), but that would be one avenue to explore to add support for Swift native components. Maybe all of the requirements to support props, state updates and event emitters from Swift have been met and we could just expose it to Swift.

**Comment by MikePendo** [comment]
We are currently working on supporting our SDK for the new Arch [Pendo](https://github.com/pendo-io/pendo-mobile-ios)
@kelset any chance u can include us in the list?
Thanks

**Reply by MikePendo** [reply]
@kelset 
Do u mind if I update it?

**Reply by cortinico** [reply]
Lorenzo is on vacation till January. Please update the table following the current structure

**Reply by kelset** [reply]
@MikePendo any links to any open issues/PRs where people can follow along/check in on status in the sdk repo?

**Reply by MikePendo** [reply]
No, unfortunately we r not open source. As a first step we are verifying that our SDK is supported by enabling Turbo modules in the apps, after that we will add the support for new bridging of Turbo Modules. Currently we have some issues with Fabric(I cant supply accurate dates about that)

**Comment by matinzd** [comment]
First `lottie-react-native` release candidate version is now available for testing which supports new architecture both on iOS and Android.
We are still investigating some bugs facing in android but hopefully it will be available on stable channel soon. 

https://www.npmjs.com/package/lottie-react-native/v/6.0.0-rc.1

Shoutout to @alfonsocj and @AlexanderEggers for helping us to get it out.

**Reply by cortinico** [reply]
Thanks you all for driving this forward! Super excited for this

**Comment by BraveEvidence** [comment]
I just created a video on how to Display qr code in react native's new architecture using fabric without 3rd party lib for iOS and Android

https://www.youtube.com/watch?v=SfWKPIVoS8w

https://www.youtube.com/watch?v=sF6PSazknq8

Someone can make an open source lib if interested

**Comment by BraveEvidence** [comment]
Scan qrcode/barcode with CameraX/MLKit in react native's new architecture fabric for android(Kotlin) and  for iOS without 3rd party lib(AVCapture)

https://www.youtube.com/watch?v=sE0MEyFA7E4
https://www.youtube.com/watch?v=q4IiUIARANM

**Comment by pushpender-singh-ap** [comment]
Hey @cortinico 

This lib support react native new architecture fabric https://github.com/pushpender-singh-ap/react-native-scanner

**Comment by RonRadtke** [comment]
https://www.npmjs.com/package/react-native-blob-util is now fully supporting the new aritechture. Last known bug on iOS is fixed for now.
Thanks for the help!

**Comment by GZLiew** [comment]
I would like to monitor live video streaming player like VLC. Here is one I found https://github.com/razorRun/react-native-vlc-media-player/issues/149

**Comment by matinzd** [comment]
React Native Lottie v6 is now out and it supports the new architecture. 

Release: https://github.com/lottie-react-native/lottie-react-native/releases/tag/v6.0.0

@kelset @cortinico

**Reply by cortinico** [reply]
Amazing stuff!

**Comment by ngocle2497** [comment]
Hey @cortinico ,
react-native-keys is in progress [PR](https://github.com/numandev1/react-native-keys/pull/37)

**Comment by ospfranco** [comment]
[op-sqlite](https://github.com/OP-Engineering/op-sqlite) is compatible with old-arch and new-arch but not bridgeless. Should bridgeless get its own category now?

[op-sqlcipher](https://github.com/OP-Engineering/op-sqlcipher) does not support new-arch yet, but it will soon, only need to merge op-sqlite into it.

**Reply by kelset** [reply]
> Should bridgeless get its own category now?

That's a good question. @cortinico WDYT?

**Reply by cipolleschi** [reply]
No, I personally don't think so. The dichotomy of New Architecture (Bridged) / New Architecture (Bridgeless) is temporary for 0.73.
From 0.74 onwards, we expect to have only `Old Architecture` / `New Architecture (Bridgeless)` (unless something else major happens that we can't foresee now)

**Reply by cortinico** [reply]
Basically what @cipolleschi 

If there are libraries that are New Architecture compatible AND are not working with Bridgeless, we should know it ASAP as we can potentially re-implement some Bridge-only APIs to work in Bridgeless mode. We want to avoid a 3-state world with Old/New (Bridge)/New (Bridgeless) as it's just too confusing

**Comment by sohayb** [comment]
Hello,
we're using the `react-native@0.72.8`, we added support for the new architecture, all is good (after overcoming some hurdles with code generation). We still can't roll it out though, as one feature depends on taking screenshots, we're using [react-native-view-shot](https://github.com/gre/react-native-view-shot), which does not support the new arch. There's already an issue opened for it #[452](https://github.com/gre/react-native-view-shot/issues/452) and #[469](https://github.com/gre/react-native-view-shot/issues/469).
Another alternative would be `react-native-skia`, which *does* support the new architecture, but for the screenshot feature.

**Comment by brentvatne** [comment]
Hi folks! We're in the home stretch here! We're making a big push for New Architecture support with React Native 0.74. For updates and info on how to help: https://github.com/reactwg/react-native-new-architecture/discussions/167. Let's continue discussions there.

**Comment by michalchudziak** [comment]
Hey! `@react-native-community/geolocation` also supports the new architecture 😀 https://github.com/michalchudziak/react-native-geolocation
